grpc/src/test/java/com/linecorp/armeria/internal/client/grpc/GrpcClientUnwrapTest.java

```diff
@@ -37,8 +38,8 @@ void test() {
         final TestServiceBlockingStub client =
                 Clients.builder("gproto+http://127.0.0.1:1/")
                        .decorator(LoggingClient.newDecorator())
-                       .decorator(RetryingClient.newDecorator(
-                               (ctx, cause) -> CompletableFuture.completedFuture(RetryDecision.noRetry())))
+                       .decorator(RetryingClient.newDecorator(RetryRule.of(
+                               (ctx, cause) -> CompletableFuture.completedFuture(RetryDecision.noRetry()))))
```

Comment:
Hmm, This seems an API level breaking changes.

Reply:
Yeah my bad.
I did this to disambiguate between the old newDecorator call that takes RetryRule, and the new one that takes RetryConfigMapping. Problem is, both are functional interfaces with two arguments. Unfortunately Java fails to infer the overload based on the type of the returned value of the lambda. The only way to keep the new overload is for callers to disambiguate by explicitly typing the lambda args, but as you said this would break existing user code.

The solution I went with is to avoid overloading for the new versions of builder() and newDecorator() that take a mapping. Instead, the new versions I added now have a different names: builderWithMapping() and newDecoratorWithMapping().

Reply:
> the new versions I added now have a different names: builderWithMapping() and newDecoratorWithMapping().

That sounds good to me. 👍 It is the most important thing that avoids breaking changes.